src/models/shelf/spec/encoding.ts

```diff
@@ -112,9 +112,8 @@ export function fromValueQuery(encQ: ValueQuery): ShelfValueDef {
   if (isWildcard(encQ.value)) {
     throw Error('Voyager does not support wildcard value');
   }
-  // TODO: return value depending on channel
   return {
-    value: encQ.value
+    value: encQ.value // TODO: when value is undefined
```

Comment:
Why don't you just do it?    It's pretty simple -- just return undefined.  

Note that everytime you push, github will send me an email for this PR since I create it. 
If you do not intend to get feedback, you might want to limit your mini-push and push a series of atomic commits when it's more ready for feedback. ;)

Reply:
Ok, will push less often, sorry... Thanks for letting me know :)